Assets/Character/Scripts/PlayerStateController.cs

```diff
@@ -135,34 +150,85 @@ void MoveStateUpdate()
 		{
 			ChangeState(Constants.STATE_IDLE);
 		}
-
-		if (_facingRight && !Input.GetKey(KeyCode.RightArrow))
+		else if (_facingRight && !Input.GetKey(KeyCode.RightArrow))
 		{
 			ChangeState(Constants.STATE_IDLE);
 		}
+		else if (PressedMidAttack())
+		{
+			ChangeState(Constants.STATE_MID_ATTACK);
+		}
+		else if (PressedRoll())
+		{
+			ChangeState(Constants.STATE_ROLL);
+		}
 	}
 
-	void RollStateUpdate()
+	void FlipSpriteRight()
 	{
-		ChangeState(Constants.STATE_IDLE);
+		_spriteRenderer.flipX = false;
+		_facingRight = true;
 	}
 
-	void JumpStateUpdate()
+	void FlipSpriteLeft()
 	{
-		ChangeState(Constants.STATE_IDLE);
+		_spriteRenderer.flipX = true;
+		_facingRight = false;
 	}
 
-	bool IsGrounded()
+	void FlipSpriteCorrectDirection()
 	{
-		float pad = .1f;
-		RaycastHit2D hit = Physics2D.Raycast(transform.position, -Vector2.up, _distToGround + pad);
-		bool result = hit.collider != null;
-		return result;
+		_spriteRenderer.flipX = !_facingRight;
 	}
 
 	bool FinishedCurrentAnimation()

```

Comment:
Maybe separate calls to animator in another class. What if we want to use the same animator calls in an enemy or a boss that also has a state machine